app/util/util.ts

```diff
@@ -107,4 +109,27 @@ const createBuyNBTCTxn = (
 	return txn;
 };
 
-export { sendTxn, trimAddress, createBuyNBTCTxn };
+const calculateTotalGasFee = async (
+	transaction: Transaction,
+	client: SuiClient,
+	mistAmount: bigint,
+	totalBalance: string,
+) => {
+	const transactionBytes = await transaction.build({ client: client });
+	const dryRunResult = await client.dryRunTransactionBlock({
+		transactionBlock: transactionBytes,
+	});
+	if (dryRunResult?.effects?.gasUsed) {
+		const { computationCost, storageCost, storageRebate } = dryRunResult.effects.gasUsed;
+		let totalGasFee = BigInt(computationCost) + BigInt(storageCost) - BigInt(storageRebate);
+		// keep buffer balance in case user try to use all max balance
+		const bufferBalanceInMist = BigInt(BUFFER_BALANCE);
```

Comment:
you should change the type of BUFFER_BALANCE to be `bigint` - just add `n` after the number, eg `10n`